perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AuditMessagesManagerBlImpl.java

```diff
@@ -69,6 +76,22 @@ public Paginated<AuditMessage> getMessagesPage(PerunSession perunSession, Messag
 		return getAuditMessagesManagerImpl().getMessagesPage(perunSession, query);
 	}
 
+	@Override
+	public List<String> findAllPossibleEvents(PerunSession sess) {
+		try {
+			List<String> events = ClassPath.from(ClassLoader.getSystemClassLoader())
+					.getAllClasses()
+					.stream()
+					.filter(clazz -> clazz.getName().matches("^cz\\.metacentrum\\.perun\\.audit\\.events\\.[^.]+\\.[A-Za-z]+$"))
```

Comment:
Feel like you could use ServiceLoader here to do it with less code and without regexp